anomalib/data/utils/boxes.py

```diff
@@ -0,0 +1,81 @@
+"""Helper functions for processing bounding box detections and annotations."""
+
+from typing import List, Tuple
+
+import torch
+from torch import Tensor
+
+from anomalib.utils.cv import connected_components_cpu, connected_components_gpu
+
+
+def masks_to_boxes(masks: Tensor) -> List[Tensor]:
+    """Convert a batch of segmentation masks to bounding box coordinates.
+
+    Args:
+        masks (Tensor): Input tensor of shape (B, 1, H, W), (B, H, W) or (H, W)
+
+    Returns:
+        List[Tensor]: A list of length B where each element is a tensor of shape (N, 4) containing the bounding box
+            coordinates of the objects in the masks in xyxy format.
+    """
+    masks = masks.view((-1, 1) + masks.shape[-2:])  # reshape to (B, 1, H, W)
+    masks = masks.float()
+
+    if masks.is_cuda:
+        batch_comps = connected_components_gpu(masks).squeeze(1)
+    else:
+        batch_comps = connected_components_cpu(masks).squeeze(1)
```

Comment:
A bit opinionated, but how about this? You don't need to accept this change.
```suggestion
    connected_components = connected_components_gpu if masks.is_cuda else connected_components_cpu
    batch_comps = connected_components(masks).squeeze(1)
```

Reply:
mypy was causing some trouble after I committed this change, so I reverted it.